### PR TITLE
buffs tinfoil hat

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request

tinfoil hat is now a ghetto mindshield. it now does the following:
-blocks implant detection: if you have a mindshield, wearing a tinfoil hat obscures it!
-blocks conversion by gang pen
-blocks ethereal dance grenade
-blocks brainwashing implants via implanter machine (the surgery still works, because the hat must be taken off anyways. The implants will cause hefty brain damage on attempted insertion)
-blocks abductor baton's stun function
-blocks both versions of mindswap
-beepsky hates tinfoil hat wearers as much as he hates wizards
-blocks being added to a slimelink
-blocks bolt of possesion
-blocks hivemind conversion
-blocks all forms of telepathy
-blocks voice of god effects
-blocks cult conversion

## Why It's Good For The Game
tinfoil hat inflicts one of the heftiest phobias in the game. Even with the phobia rework pr, it's not worth wearing.

this makes the cost worth it. even so, a tinfoil hat is quite easy to remove by anyone but the wearer.

## Changelog
:cl:
tweak: mass mindswap now checks antimagic
tweak: tinfoil plasmaman helmet offers the same protection as other envirohelms
balance: tinfoil hat is now way stronger
/:cl:

### CODE REVIEW

File: code/game/objects/effects/mines.dm [hunk 1]
```diff
@@ -121,7 +121,7 @@
 	if(target)
 		if(!(target && HAS_TRAIT(target, TRAIT_MINDSHIELD)))
 			triggermine(target)
-		if(smartmine == 0)
+		if(smartmine == 0 || !istype(target.get_item_by_slot(SLOT_HEAD), /obj/item/clothing/head/foilhat)) //tinfoil hat prevents detection of implants
```
Comment (PowerfulBacon) [inline code/game/objects/effects/mines.dm]: Not wearing a tinfoil hat triggers the mine? Even if you are mindshielded?

File: code/game/objects/items/grenades/discogrenade.dm [hunk 2]
```diff
@@ -108,6 +108,10 @@
 		M.show_message("<span class='warning'>You resist your inner urges to break out your best moves.</span>", 2)
 		M.set_drugginess(5)
 		return
+	if(istype(M.get_item_by_slot(SLOT_HEAD), /obj/item/clothing/head/foilhat))
+		to_chat(M, "<span class = 'userdanger'>THOSE GLOW-IN-THE-DARK NANOTRASEN LIGHTBULBS WON'T CORRUPT ME WITH THEIR QUEER AGENDA</span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/grenades/discogrenade.dm]: > QUEER AGENDA

Comment (zeskorion) [reply code/game/objects/items/grenades/discogrenade.dm]: yes.

Comment (PowerfulBacon) [reply code/game/objects/items/grenades/discogrenade.dm]: No

Comment (zeskorion) [reply code/game/objects/items/grenades/discogrenade.dm]: You got anything better

Comment (Mick1299) [reply code/game/objects/items/grenades/discogrenade.dm]: Xenophilic?

Comment (zeskorion) [reply code/game/objects/items/grenades/discogrenade.dm]: i dunnoooo....

Comment (ghost) [reply code/game/objects/items/grenades/discogrenade.dm]: Shouldn't be any politics in ss13 and if there is then everything should be allowed.

Comment (TeomanTheGreat) [reply code/game/objects/items/grenades/discogrenade.dm]: Lmao queer agenda

**Comment by PowerfulBacon** [comment]
Putting to vote, remove the queer part for merge if accepted

File: code/game/objects/items/grenades/discogrenade.dm [hunk 3]
```diff
@@ -108,6 +108,10 @@
 		M.show_message("<span class='warning'>You resist your inner urges to break out your best moves.</span>", 2)
 		M.set_drugginess(5)
 		return
+	if(istype(M.get_item_by_slot(SLOT_HEAD), /obj/item/clothing/head/foilhat))
+		to_chat(M, "<span class = 'userdanger'>THOSE GLOW-IN-THE-DARK NANOTRASEN LIGHTBULBS WON'T CORRUPT ME WITH THEIR AGENDA</span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/grenades/discogrenade.dm]:
```suggestion
		to_chat(M, "<span class = 'userdanger'>THOSE GLOW-IN-THE-DARK NANOTRASEN LIGHTBULBS WON'T CORRUPT ME WITH THEIR AGENDA!</span>")
```

File: code/modules/mob/living/carbon/human/species_types/jellypeople.dm [hunk 4]
```diff
@@ -596,6 +596,9 @@
 		return FALSE
 	if(HAS_TRAIT(M, TRAIT_MINDSHIELD)) //mindshield implant, no dice
 		return FALSE
+	if(istype(M.get_item_by_slot(SLOT_HEAD), /obj/item/clothing/head/foilhat))
+		to_chat(M, "<span class='danger'>[slimelink_owner.real_name]'s no-good syndicate mind-slime is blocked by your protective headgear!.</span>")
```
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/species_types/jellypeople.dm]:
```suggestion
		to_chat(M, "<span class='danger'>[slimelink_owner.real_name]'s no-good syndicate mind-slime is blocked by your protective headgear!</span>")

```

**Review by PowerfulBacon — COMMENTED** [review]
Lightbulbs works, final check up

File: code/game/objects/items/implants/implantchair.dm [hunk 5]
```diff
@@ -191,6 +191,13 @@
 		log_game("[key_name(user)] set brainwash machine objective to '[objective]'.")
 	if(HAS_TRAIT(C, TRAIT_MINDSHIELD))
 		return FALSE
+	if(ishuman(C))
+		var/mob/living/carbon/human/H = C
+		if(istype(H.get_item_by_slot(SLOT_HEAD), /obj/item/clothing/head/foilhat))
+			to_chat(H, "<span class ='userdanger'>Your trusty tinfoil hat shorts out the implant as it plunges into your skull!</span>")
+			H.adjustOrganLoss(ORGAN_SLOT_BRAIN, 75)
+			H.emote("scream")
+		return TRUE
```
Comment (PowerfulBacon) [inline code/game/objects/items/implants/implantchair.dm]: doesn't work on humans anymore

**Review by PowerfulBacon — APPROVED** [review]
cool and good

**Pull request merged**